awx/api/serializers.py

```diff
@@ -4893,6 +4893,7 @@ class Meta:
             'created',
             'modified',
             'last_seen',
+            'health_check_started',
```

Comment:
Since this field is used so infrequently (and so specifically to user-initiated or execution-node triggers) it might be better to _hide_ this field by making it `editable=False`, not showing it in the serializer, and replacing it with a field like `health_check_in_progress`, which would only give `True` if a user request is still in progress, or an execution node check was in progress (probably in response to some user-initiated action).

Control plane nodes still have health checks, and it's unreasonable to apply any of this to that.

Reply:
It is already `editable=False`.

Reply:
> Control plane nodes still have health checks, and it's unreasonable to apply any of this to that.

@AlanCoding Why?  The instance health check endpoint can trigger the control plane health check.  In fact, it is the *only* entry point into `cluster_node_health_check` as far as I can see.

My intention is that any process, automated or manual, that would write to `last_health_check` ought to get a corresponding write beforehand to `health_check_started`.

Reply:
> ... not showing it in the serializer, and replacing it with a field like `health_check_in_progress` ...

@kialam 's changes are effectively already doing this on the frontend, but my leanings are that if someone is peeping into the output of the API they actually care about the details and we shouldn't try to abstract it away into a less informative "field".

Reply:
> My intention is that any process, automated or manual, that would write to last_health_check ought to get a corresponding write beforehand to health_check_started.

That's a bad idea, because IIRC `cluster_node_heartbeat` updates `last_health_check` in a periodic task every 20 seconds, and you can't consolidate those saves (by statement of their intended function).

If you run a manual health check for a control node, it's not even clear whether the updated health check data came from the task you triggered _or the periodic task_, which is why this entire function is nonsense for the control plane and should be disabled for those node types. Nonetheless, you don't want to selectively hide fields in the serializer. At least if added `health_check_in_progress` as a boolean, it can be documented to only cover manual health checks and somewhat honestly return False for control plane nodes.

Reply:
Ok, I have removed the writes to the new field for automated processes.  It only writes from the API endpoint now.

Reply:
you're still showing the field in the serializer, and I still think that could be confusing in cases.